examples/dialogs/dialogs/app.py

```diff
@@ -90,12 +90,53 @@ def action_save_file_dialog(self, widget):
         except ValueError:
             self.label.text = "Save file dialog was canceled"
 
+    def action_open_secondary_window(self, widget):
+        # Problems:
+        # - Setting secondary window's `app` manually
```

Comment:
This is a good point - how do we tell the window about the app?

The MainWindow gets assigned to the app when you set `self.main_window = ...`. I wonder if we should do a similar thing for other windows, and require that they are explicitly added to the app:

    self.windows += toga.Window(...)

or:

    self.windows.add(toga.Window(...))


and raise an error if you attempt to call `show()` on a window that doesn't have an app assigned. 

In this world, `TogaApp.windows` would be a set-like object that has add/remove methods (including support for the + and += operators); adding a window to `TogaApp.windows` would assign the window to the app.

Does that make sense as an API?

Reply:
I've added a possible implementation of self.windows, could you take a look?
The example doesn't count windows correctly because the handling of window on_close is incorrect at the moment.